pkg/clients/compute.go

```diff
@@ -25,23 +25,29 @@ import (
 	"github.com/gophercloud/gophercloud/openstack/compute/v2/extensions/availabilityzones"
 	"github.com/gophercloud/gophercloud/openstack/compute/v2/flavors"
 	"github.com/gophercloud/gophercloud/openstack/compute/v2/servers"
+	"github.com/gophercloud/gophercloud/openstack/utils"
 	"github.com/gophercloud/utils/openstack/clientconfig"
 	uflavors "github.com/gophercloud/utils/openstack/compute/v2/flavors"
 
 	"sigs.k8s.io/cluster-api-provider-openstack/pkg/metrics"
 )
 
 /*
-NovaMinimumMicroversion is the minimum Nova microversion supported by CAPO
-2.60 corresponds to OpenStack Queens
+NovaSupportedVersions is the list of Nova microversion supported by CAPO
+2.60 corresponds to OpenStack Queens and 2.53 to OpenStack Pike.
 
 For the canonical description of Nova microversions, see
 https://docs.openstack.org/nova/latest/reference/api-microversion-history.html
 
-CAPO uses server tags, which were added in microversion 2.52.
+CAPO uses server tags, which were first added in microversion 2.26 and then refined
+in 2.52 so it is possible to apply them when creating a server (which is what CAPO does).
 CAPO supports multiattach volume types, which were added in microversion 2.60.
 */
-const NovaMinimumMicroversion = "2.60"
+var NovaSupportedVersions = []*utils.Version{
+	{ID: "2.1", Priority: 10, Suffix: "/v2.1/"},   // Initial microversion release, same as specifying no version
+	{ID: "2.53", Priority: 20, Suffix: "/v2.53/"}, // Maximum in Pike
+	{ID: "2.60", Priority: 30, Suffix: "/v2.60/"}, // Maximum in Queens
```

Comment:
we forced to 2.60 at commit 2df3778e989a227a542201ee7873b8d51d7f9c8d
so I think at some specific call we should use that version ,but I didn't see we have such negotiation 
in the following code, can you point me where's the logic that choose 2.60 after the change?Thanks

Reply:
The version is chosen [here](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/c38aa7f44e28ab6316033555ba25d46a3b1455f0/pkg/clients/compute.go#L81). It will check what versions are supported by the server and then pick a version from this list that is supported. I also made sure to put highest priority for 2.60 so this is what should be preferred if it is available.